### PR TITLE
worked on approval flow

### PR DESCRIPTION
**Pull request by anilkumarkammalapalli**

### CODE REVIEW

File: src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java [hunk 1]
```diff
@@ -61,53 +61,64 @@ public class TaxonomyServiceImpl implements WfServiceHandler {
...
-                    term.put(Constants.TERM,requestMap);
-                    request.put(Constants.REQUEST,term);
-                    String URI = constructTermUpdateURI(id, category);
-                    logger.info("printing URI For Term Update {} ", URI);
-                    Map<String, Object> response = requestService.fetchResultUsingPatch(URI,request, null);
-                    String responseCode = (String) response.get(Constants.RESPONSE_CODE);
-                    if (!responseCode.equals(Constants.OK)){
-                        logger.error("Unable To Update Term Status");
+                    StringBuilder termRead = constructReadTermApiURI(id, category);
+                    Map<String, Object> termResponse = (Map<String, Object>) requestService.fetchResultUsingGet(termRead);
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java]: just have one method named - readTermObject and pass the required details to it... it should construct url and call requestService

File: src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java [hunk 2]
```diff
@@ -61,53 +61,64 @@ public class TaxonomyServiceImpl implements WfServiceHandler {
...
+                    String responseForTermRead = (String) termResponse.get(Constants.RESPONSE_CODE);
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java]: Do not create variables unnecessarily... you can write the following... 
if (Constants.OK.equalsIgnoreCase((String) termResponse.get(Constants.RESPONSE_CODE)))

File: src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java [hunk 3]
```diff
@@ -61,53 +61,64 @@ public class TaxonomyServiceImpl implements WfServiceHandler {
...
+                    if (responseForTermRead.equals(Constants.OK)) {
+                        Map<String, Object> resultMap = (Map<String, Object>) termResponse.get(Constants.RESULT);
+                        Map<String, Object> resultTerm = (Map<String, Object>) resultMap.get(Constants.TERM);
+                        String currentApprovalStatus = (String) resultTerm.get(Constants.APPROVAL_STATUS);
+                        if (!currentApprovalStatus.equals(Constants.Live)) {
+                            String currentStatus = wfStatusEntityV2.getCurrentStatus();
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java]: currentStatus is unnecessary variable

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java (hunk 1 above)
Comment (anilkumarkammalapalli) [reply src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java]: yes i changed the code according to your suggestions please review.

File: src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java [hunk 4]
```diff
@@ -149,4 +157,31 @@ private List<HashMap<String, Object>> getUpdateFieldValues(WfStatusEntityV2 stat
         }
         return updateFieldValuesList;
     }
+
+    private StringBuilder constructReadTermApiURI(String term, String category) {
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java]: this method is not called from anywhere ?

File: src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java [hunk 5]
```diff
@@ -116,20 +124,20 @@ public void processMessage(WfRequest wfRequest) {
...
         return uri;
     }
 
     private StringBuilder constructPublishFrameworkURI() {
         StringBuilder builder = null;
-        if (!StringUtils.isEmpty(frameworkId)){
+        if (!StringUtils.isEmpty(frameworkId)) {
             builder = new StringBuilder();
-            UriComponents uriComponents = UriComponentsBuilder.fromUriString(host + PUBLISH_FRAMEWORK_URI.replace(Constants.ID,frameworkId)).build();
+            UriComponents uriComponents = UriComponentsBuilder.fromUriString(host + PUBLISH_FRAMEWORK_URI.replace(Constants.ID, frameworkId)).build();
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java]: UriComponentBuilder is unnecessary -- you can directly use the StringBuilder instead.

File: src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java [hunk 6]
```diff
@@ -149,4 +157,31 @@ private List<HashMap<String, Object>> getUpdateFieldValues(WfStatusEntityV2 stat
...
+        }
+        return builder;
+    }
+
+    private Map<String, Object> readTermObject(String term, String category) {
+        StringBuilder builder = null;
+        Map<String, Object> termResponse = null;
+        if (!StringUtils.isEmpty(term) && !StringUtils.isEmpty(category)) {
+            builder = new StringBuilder();
+            UriComponents uriComponents = UriComponentsBuilder.fromUriString(host + termReadURI.replace(Constants.ID, term))
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java]: UriComponentBuilder is unnecessary -- you can directly use the StringBuilder instead.

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java (hunk 4 above)
Comment (anilkumarkammalapalli) [reply src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java]: yes i will remove it is not requried.

File: src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java (hunk 6 above)
Comment (anilkumarkammalapalli) [reply src/main/java/org/sunbird/workflow/service/impl/TaxonomyServiceImpl.java]: yes i did changes according to your suggestion please review.

**Review by karthik-tarento — APPROVED** [review]

**Pull request merged**